test/common/http/http1/codec_impl_test.cc

```diff
@@ -1537,6 +1537,70 @@ TEST_F(Http1ServerConnectionImplTest, UpgradeRequestWithNoBody) {
   codec_->dispatch(buffer);
 }
 
+TEST_F(Http1ServerConnectionImplTest, ConnectRequestNoContentLength) {
+  initialize();
+
+  InSequence sequence;
+  NiceMock<MockRequestDecoder> decoder;
+  EXPECT_CALL(callbacks_, newStream(_, _)).WillOnce(ReturnRef(decoder));
+
+  EXPECT_CALL(decoder, decodeHeaders_(_, false)).Times(1);
+  Buffer::OwnedImpl buffer("CONNECT / HTTP/1.1\r\n\r\n");
```

Comment:
I'm not very familiar with CONNECT, but according to https://tools.ietf.org/html/rfc7231#section-4.3.6 it says that CONNECT requires the authority form. Should we be allowing a request like this?

Reply:
No, we shouldn't.  It gets by because of the if(is_connect) in ServerConnectionImpl::handlePath.
Fixing that requires fixing the URL parser to handle connect requests, and the audit of all the Path() calls in the code base.  Ok with this and the updated TODO or would you prefer to land this code and the Path() work together?

Reply:
I was thinking we could at least fix the codec in this PR and then put in the synthetic `\` before dispatch into the HCM? WDYT?